cloud/resource_cloud_connection.go

```diff
@@ -0,0 +1,258 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	cloudv1alpha1 "github.com/streamnative/cloud-api-server/pkg/apis/cloud/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func resourceCloudConnection() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCloudConnectionCreate,
+		ReadContext:   resourceCloudConnectionRead,
+		UpdateContext: resourceCloudConnectionUpdate,
+		DeleteContext: resourceCloudConnectionDelete,
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+			oldOrg, _ := diff.GetChange("organization")
+			oldName, _ := diff.GetChange("name")
+			if oldOrg.(string) == "" && oldName.(string) == "" {
+				// This is create event, so we don't need to check the diff.
+				return nil
+			}
+			if diff.HasChange("name") ||
+				diff.HasChanges("organization") ||
+				diff.HasChanges("cloud_connection_name") ||
+				diff.HasChanges("region") ||
+				diff.HasChanges("network_id") ||
+				diff.HasChanges("network_cidr") {
+				return fmt.Errorf("ERROR_UPDATE_CLOUD_CONNECTION: " +
+					"The cloud connection does not support updates, please recreate it")
+			}
+			return nil
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				organizationInstance := strings.Split(d.Id(), "/")
+				_ = d.Set("organization", organizationInstance[0])
+				_ = d.Set("name", organizationInstance[1])
+				err := resourceCloudConnectionRead(ctx, d, meta)
+				if err.HasError() {
+					return nil, fmt.Errorf("import %q: %s", d.Id(), err[0].Summary)
+				}
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+		Schema: map[string]*schema.Schema{
+			"organization": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  descriptions["organization"],
+				ValidateFunc: validateNotBlank,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  descriptions["name"],
+				ValidateFunc: validateNotBlank,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  descriptions["type"],
```

Comment:
Please add a description to provider.go file for type

Reply:
Added